cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+import itertools
+from typing import Sequence, Dict
+from cirq import circuits, devices, ops, protocols, sim, study, value
+import numpy
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiExperimentResult(object):
+    """Class for storing and plotting data from a Rabi oscillation
+    experiment.
+    """
+
+    def __init__(self, num_shots: int, rabi_angles: Sequence[float],
+                 excited_state_pops: Sequence[float]):
+        """
+        Args:
+            num_shots: The number of repetitions of the circuit for each Rabi
+              angle.
+            rabi_angles: The rotation angles of the qubit around the x-axis
+              of the Bloch sphere.
+            excited_state_pops: The corresponding probabilities that the
+            qubit is in the excited state.
+        """
+        self._num_shots = num_shots
+        self._rabi_angles = rabi_angles
+        self._excited_state_pops = excited_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
```

Comment:
Docstring describing the data.

In the future we may want to go even more granular than probabilities and do the individual samples. People may want to analyze it for e.g. bursts of too-many-1s. For now this is good.

Reply:
Done